### PR TITLE
Created an Issue Filing dialog

### PR DESCRIPTION
**Pull request by waabid**
#### Description of changes

Created a dialog for the new issue filing experience to be used for any scenario when user has not setup their option/service.

The content of the file issue dialog is not a part of this PR as it is a separate component to be used in settings and the dialog.

The UI changes for buttons was validated/verified with Chelsea.

#### Pull request checklist

- [n/a] Addresses an existing issue: Fixes #0000
- [x] Added relevant unit test for your changes. (`npm run test`)
- [x] Verified code coverage for the changes made. Check coverage report at: `<rootDir>/test-results/unit/coverage`
- [n/a] Ran precheckin (`npm run precheckin`)
- [x] Added screenshots/GIFs for UI changes.

image
image

### CODE REVIEW

**Comment by waabid** [comment]
Making it not to merge as I'm hoping to re-factor tests.

File: src/tests/unit/tests/DetailsView/components/action-and-cancel-buttons-component.test.tsx [hunk 1]
```diff
@@ -16,28 +16,31 @@ describe('ActionAndCancelButtonsComponent', () => {
...
+                <div className="button action-cancel-button-col">
+                    <DefaultButton text="Cancel" onClick={props.cancelButtonOnClick} />
+                </div>
+                <div className="button action-cancel-button-col">
                     <DefaultButton
                         primary={true}
                         text={props.primaryButtonText}
                         onClick={props.primaryButtonOnClick}
                         disabled={props.primaryButtonDisabled}
+                        href={sampleHref}
```
Comment (smoralesd) [inline src/tests/unit/tests/DetailsView/components/action-and-cancel-buttons-component.test.tsx]: does it make sense to update this test to use snapshot?

File: src/tests/unit/tests/DetailsView/components/issue-filing-dialog.test.tsx [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+        createBugFilingUrlMock
+            .setup(createBugFilingUrl => createBugFilingUrl(selectedServiceData, selectedBugDataStub))
+            .verifiable(Times.exactly(2));
+        telemetryCallbackMock.setup(telemetryCallback => telemetryCallback(eventStub)).verifiable();
+
+        const testSubject = shallow(<IssueFilingDialog {...props} />);
+        expect(testSubject.getElement()).toMatchSnapshot('Opened Dialog');
+
+        const actionCancelButtons = testSubject.find(ActionAndCancelButtonsComponent);
+        actionCancelButtons.prop('primaryButtonOnClick')(eventStub);
```
Comment (smoralesd) [inline src/tests/unit/tests/DetailsView/components/issue-filing-dialog.test.tsx]: should the click on buttons be on separated test? Why are we not using `.simulate('click', ...)` to "click" this buttons?

Comment (waabid) [reply src/tests/unit/tests/DetailsView/components/issue-filing-dialog.test.tsx]: Agreed about concerns; hence why I made it do not merge 😄.

Regarding stimulate: I think since it is a shallow render, I might not be able to get the elements (cancel and file issue specifically) below actioncancelbutton component. If that is not true, I can change that.

File: src/tests/unit/tests/DetailsView/components/action-and-cancel-buttons-component.test.tsx (hunk 1 above)
Comment (waabid) [reply src/tests/unit/tests/DetailsView/components/action-and-cancel-buttons-component.test.tsx]: Yes, but I didn't want to do this in the PR for this.

**Comment by smoralesd** [comment]
I'm confused with the screenshot as it shows the "Add failure instance" from assessment not from the "file issue" button... or am I looking at this from the wrong side?

File: src/DetailsView/components/issue-filing-dialog.tsx [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+            onDismiss={onClose}
+        >
+            <DialogFooter>
+                <ActionAndCancelButtonsComponent
+                    isHidden={false}
+                    primaryButtonDisabled={selectedBugFilingService.isSettingsValid(selectedBugFilingServiceData)}
+                    primaryButtonOnClick={bugFileTelemetryCallback}
+                    cancelButtonOnClick={onClose}
+                    primaryButtonHref={selectedBugFilingService.createBugFilingUrl(selectedBugFilingServiceData, selectedBugData)}
+                    primaryButtonText={'File issue'}
```
Comment (smoralesd) [inline src/DetailsView/components/issue-filing-dialog.tsx]: I guess we may want to get this string (`'File issue'`) from the service but I'm fine if we do that on a different PR

File: src/DetailsView/components/issue-filing-dialog.tsx [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+    selectedBugFilingServiceData: BugServiceProperties;
+    bugFileTelemetryCallback: (ev: React.SyntheticEvent) => void;
+    onClose: (ev: React.SyntheticEvent) => void;
+}
+
+export interface IssueFilingDialogDeps {}
+
+const titleLabel = 'Specify issue filing location';
+
+export const IssueFilingDialog = NamedSFC<IssueFilingDialogProps>('IssueFilingDialog', props => {
```
Comment (smoralesd) [inline src/DetailsView/components/issue-filing-dialog.tsx]: Two questions:
- This is not being use anywhere at this moment, right?
- I don't see the different issue filing options being rendered here, just the buttons to 'cancel' and 'file issue'. Am I missing something?

File: src/DetailsView/components/issue-filing-dialog.tsx (hunk 3 above)
Comment (waabid) [reply src/DetailsView/components/issue-filing-dialog.tsx]: This isn't coming from the service since we aren't don't that as a part of this feature as of yet.

File: src/DetailsView/components/issue-filing-dialog.tsx (hunk 4 above)
Comment (waabid) [reply src/DetailsView/components/issue-filing-dialog.tsx]: 1. Correct, not yet.
2. Yes, because the content/container for that content is a different control (since that will be re-used in settings as well).

**Comment by waabid** [comment]
> I'm confused with the screenshot as it shows the "Add failure instance" from assessment not from the "file issue" button... or am I looking at this from the wrong side?

I made some modifications to the UI of the action button component. Specifically, some of the styling, which was partially inadequate/broken (zoom in enough and it looks bad/broken) and partially we want to change how it looks previously.

Previously, it used to say <Action> <Cancel> and now it is <Cancel> and <Action>. Since the action cancel button is being used in the add failure panel as well, I figured I'd include it as well (as mentioned earlier, this was verified with Chelsea).
